backend/src/crd/controller/viewer/reconciler/reconciler.go

```diff
@@ -83,6 +83,16 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 	}
 	glog.Infof("Got instance: %+v", view)
 
+	if len(view.Spec.TensorboardSpec.TensorflowImage) == 0 {
+		if err := r.Client.Delete(context.Background(), view); err != nil {
+			glog.Infof("Error in deleting viewer CRD: %+v", err)
+			return reconcile.Result{}, err
+		} else {
+			glog.Infof("Deleted viewer CRD that is missing tensorflow image.")
+			return reconcile.Result{}, nil
+		}
+	}
+
```

Comment:
why adding this logic instead of using CRD built-in openapi validation
https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#specifying-a-structural-schema

Reply:
Our current implementation doesn't seem to support structural schema yet. After a quick browse of the doc, we'll need to first add/apply a schema yaml. And then all the viewer CRD yamls created later can be validated according to the specifications in the schema yaml. I'll do another PR to upgrade our current viewer crd implementation to structural schema and let this one focus on tensorboard version issue.

Reply:
Can we not check in the validation logic here then and add a todo?

Reply:
sure. Leave validation to when we have structural crd schema. Temporarily, I'll just use  a default version if no version is specified.